v2/mongodb-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/mongodb/templates/MongoDbUtils.java

```diff
@@ -88,17 +113,18 @@ public static Document getMongoDbDocument(String uri, String dbName, String coll
     return doc;
   }
 
-  public static TableRow getTableSchema(HashMap<String, Object> parsedMap, String userOption) {
+  public static TableRow getTableSchema(Document document, String userOption) {
     TableRow row = new TableRow();
-    if (userOption.equals("FLATTEN")) {
-      parsedMap.forEach(
+    LOG.info("Document" + document);
+    if (userOption.equals("FLATTEN") || userOption.equals("UDF")) {
```

Comment:
Could just check "FLATTEN" - what if a user wants to use a UDF with NONE userOption?

Reply:
You can use one of the three options. Not all of them together. Flatten and UDF uses the same logic to get the rows for bigquery

Reply:
Will add this suggetion and test out the templates.

Reply:
Updated